### PR TITLE
Improve wording in note helpers' rename note command

### LINKED ISSUE #513: [Note Helpers] /rename inconsistent note filenames prompt text says it's doing the opposite of what it actually does
I was told to @leomeloxp on this one.
**Describe the bug**
when using the /rename inconsistent filenames command, the prompt suggests that the note will be renamed to suit the filename.md, but in fact it does it the other way around. Quite unexpected!!

**Expected behavior**
when I choose a folder after running /rename inconsistent filenames, and ask it to rename all of my files without asking me each time, i expect it to rename the note based on the current filename, becuase that's what the prompt suggests. 

Note, however, that i'd also like a parallel function that does the exact opposite, letting me rename my filename based on the current Note Title. 

**Screenshots**
Screenshot_2024-01-10_NotePlan Beta_000563@2x
### before running the /rename inconsistent filenames command: 
screenshot of my note files in Finder:
Screenshot_2024-01-10_Finder_000576
screenshot of my note filenames the command found: 
Screenshot_2024-01-10_NotePlan Beta_000574@2x
screenshot of my note titles in NotePlan:
Screenshot_2024-01-10_NotePlan Beta_000578@2x
### after running the command: 
screenshot of my note filenames in Finder
Screenshot_2024-01-10_Finder_000580

It's obviously using the note-name to rename the filename, not renaming the note to match the filename as the command prompt suggests.

**System**
 - Device: [ Mac]
 - NotePlan Version [Version 3.9.11 (1138)]
 - Plugin Name & Version [e.g. "NoteHelpers" v0.19.0]

**Additional context**
[Discord message discussion](https://discord.com/channels/763107030223290449/1117945664500924437/1194678884550049813)

### PR DESCRIPTION
**Pull request by leomeloxp**
Fix #513

### CODE REVIEW

File: jgclark.NoteHelpers/src/helpers/renameNotes.js [hunk 1]
```diff
@@ -24,34 +24,35 @@ export async function renameNoteToTitle(note: Note, shouldPromptBeforeRenaming:
...
     const newFilename = note.rename(newPath)
     logDebug(pluginJson, `renameNoteToTitle(): ${currentFullPath} -> ${newFilename}`)
-    return
+    return true
   }
 
+  const currentFilename = currentFullPath.split('/').pop()
   const promptResponse = await showMessageYesNoCancel(`
-  Would you like to rename the note ${title} to match its filename?
+  Would you like to rename "${currentFilename}" to match the note title "${title}"?
```
Comment (leomeloxp) [inline jgclark.NoteHelpers/src/helpers/renameNotes.js:45]: The fix to the issue highlighted raised by @ariccb.

File: jgclark.NoteHelpers/plugin.json [hunk 2]
```diff
@@ -139,6 +139,14 @@
...
       "jsFunction": "listInconsistentNames"
     },
+    {
+      "name": "reset title to match filename",
+      "alias": [
+        "filenameToTitle"
+      ],
+      "description": "Resets the current note title to match its filename.",
+      "jsFunction": "filenameToTitle"
+    },
```
Comment (leomeloxp) [inline jgclark.NoteHelpers/plugin.json:149]: See code below for implementation details. This adds a new command for setting the filename (excluding extension) as the note title.

<details><summary>Filename</summary>
SCR-20240208-umum
</details> 

<details><summary>Noteplan command</summary>
SCR-20240208-umiv
</details> 

<details><summary>Confirmation prompt</summary>
SCR-20240208-umrq
</details>

File: jgclark.NoteHelpers/src/helpers/makeNoteTitleMatchFilename.js [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+export async function makeNoteTitleMatchFilename(note: Note, shouldPromptBeforeRenaming: boolean = true): Promise<boolean> {
+  const { defaultFileExtension } = DataStore
+  if (note == null || note.paragraphs.length < 1) {
+    // No note open, so don't do anything.
+    logDebug(pluginJson, 'No note open, or no content. Stopping.')
+    return false
+  }
+  if (note.type === 'Calendar') {
+    // Won't work on calendar notes
+    await showMessage('This command does not support calendar notes.')
```
Comment (jgclark) [inline jgclark.NoteHelpers/src/helpers/makeNoteTitleMatchFilename.js:16]: Or "Sorry: calendar notes cannot be renamed."

File: jgclark.NoteHelpers/src/lib/commands/renameInconsistentNames.js [hunk 4]
```diff
@@ -42,7 +42,7 @@ export async function renameInconsistentNames(): Promise<void> {
     `)
 
     const response = await showMessageYesNoCancel(
-      `Would you like to be prompted before renaming each of these notes?
+      `Would you like to be prompted before renaming each of these notes filenames?
```
Comment (jgclark) [inline jgclark.NoteHelpers/src/lib/commands/renameInconsistentNames.js:45]:
```suggestion
      `Would you like to be prompted before renaming each of these note filenames?
```

File: jgclark.NoteHelpers/src/helpers/makeNoteTitleMatchFilename.js (hunk 3 above)
Comment (jgclark) [inline jgclark.NoteHelpers/src/helpers/makeNoteTitleMatchFilename.js:16]:
```suggestion
    await showMessage('Sorry: calendar notes cannot be renamed.')
```

File: jgclark.NoteHelpers/src/helpers/renameNotes.js
Comment (jgclark) [inline jgclark.NoteHelpers/src/helpers/renameNotes.js:1]: I think somewhere in here you need to check for cases where there are valid title characters that aren't valid filename characters. I can think of at least `:`, but there may be others. And you may want to defeat on leading `.` as well.

**Review by jgclark — CHANGES_REQUESTED** [review]
Good work. I've made a few small comments, only one of which (about invalid characters) is important.
I will have to look when I'm at my desktop to see what the test problem is.

File: jgclark.NoteHelpers/src/helpers/renameNotes.js
Comment (leomeloxp) [reply jgclark.NoteHelpers/src/helpers/renameNotes.js:1]: Added a function to check for it and report an error to the user if any invalid characters are found.

I didn't really read too much into which characters should be added to the list but I at least there is something in there.

**Review by jgclark — APPROVED** [review]

**Pull request merged**